lib/webflow/client.rb

```diff
@@ -131,8 +131,8 @@ def request(path, method: :get, params: nil, data: nil)
 
       track_rate_limit(response.headers)
 
-      result = JSON.parse(response.body)
-      raise Webflow::Error.new(result) if response.code >= 400
+      result = maybe_parsed_json(response.body)
```

Comment:
this is not very declerative. the api is supposed to return a json payload in all cases. if there are exceptional cases that do not return json, this should be obvious in code. if parsing json raises an error and this is not forseable by the response-code, we should raise an error with a specific message, so that a user could retry or the like.

Reply:
The non-json response body is passed to Webflow::Error so the user receives an exception with the response body as exception message. in my case it is 'Rate limit hit'.

Reply:
this should be indicated with status code `429`

Reply:
will check =)